### PR TITLE
config: pre-create pinns directories

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

> /kind api-change
/kind bug
> /kind ci
> /kind cleanup
> /kind dependency-change
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake
> /kind other

#### What this PR does / why we need it:
as well as remove any file that exists in that spot

Signed-off-by: Peter Hunt <pehunt@redhat.com>
#### Which issue(s) this PR fixes:

should fix https://bugzilla.redhat.com/show_bug.cgi?id=1922648

#### Does this PR introduce a user-facing change?

```release-note
none
```

### CODE REVIEW

**Review by giuseppe — APPROVED** [review]
LGTM

File: internal/config/nsmgr/types.go [hunk 1]
```diff
@@ -24,6 +24,13 @@ const (
 	ManagedNamespacesNum        = 4
 )
 
+// SupportedNamespaces returns a slice of
+// the names of namespaces that CRI-O supports
+// pinning.
+func SupportedNamespaces() []NSType {
```
Comment (mrunalp) [inline internal/config/nsmgr/types.go:30]: nit: Maybe `SupportedNamespacesForPinning`?

Comment (haircommander) [reply internal/config/nsmgr/types.go]: updated as suggested

**Review by saschagrunert — APPROVED** [review]
LGTM

**Pull request merged**